frontend/src/metabase/query_builder/containers/QueryBuilder.tsx

```diff
@@ -94,96 +101,120 @@ const timelineProps = {
+  visibleTimelineEventIds: getVisibleTimelineEventIds(state),
+  xDomain: getTimeseriesXDomain(state),
+
+  result: getFirstQueryResult(state),
+  results: getQueryResults(state),
+  rawSeries: getRawSeries(state),
+
+  uiControls: getUiControls(state),
+  ...state.qb.uiControls,
+  dataReferenceStack: getDataReferenceStack(state),
+  isAnySidebarOpen: getIsAnySidebarOpen(state),
+
+  // isBookmarked: getIsBookmarked(state, props),
+  isDirty: getIsDirty(state),
+  isObjectDetail: getIsObjectDetail(state),
+  isNativeEditorOpen: getIsNativeEditorOpen(state),
+  isNavBarOpen: getIsNavbarOpen(state),
+  isVisualized: getIsVisualized(state),
+  isLiveResizable: getIsLiveResizable(state),
+  isTimeseries: getIsTimeseries(state),
+  isHeaderVisible: getIsHeaderVisible(state),
+  isActionListVisible: getIsActionListVisible(state),
+  isAdditionalInfoVisible: getIsAdditionalInfoVisible(state),
+
+  parameters: getParameters(state),
+  databaseFields: getDatabaseFields(state),
+  sampleDatabaseId: getSampleDatabaseId(state),
+
+  isRunnable: getIsRunnable(state),
+  isResultDirty: getIsResultDirty(state),
+  isMetadataDirty: isResultsMetadataDirty(state),
+
+  questionAlerts: getQuestionAlerts(state),
+  visualizationSettings: getVisualizationSettings(state),
+
+  autocompleteResultsFn: getAutocompleteResultsFn(state),
+  cardAutocompleteResultsFn: getCardAutocompleteResultsFn(state),
+
+  queryStartTime: getQueryStartTime(state),
+  nativeEditorCursorOffset: getNativeEditorCursorOffset(state),
+  nativeEditorSelectedText: getNativeEditorSelectedText(state),
+  modalSnippet: getModalSnippet(state),
+  snippetCollectionId: getSnippetCollectionId(state),
+  documentTitle: getDocumentTitle(state),
+  pageFavicon: getPageFavicon(state),
+  isLoadingComplete: getIsLoadingComplete(state),
+
+  reportTimezone: getSetting(state, "report-timezone-long"),
+
+  getEmbeddedParameterVisibility: (slug: string) =>
+    getEmbeddedParameterVisibility(state, slug),
+});
 
 const mapDispatchToProps = {
   ...actions,
   closeNavbar,
   onChangeLocation: push,
-  createBookmark: id => Bookmark.actions.create({ id, type: "card" }),
-  deleteBookmark: id => Bookmark.actions.delete({ id, type: "card" }),
+  createBookmark: (id: BookmarkId) =>
+    Bookmark.actions.create({ id, type: "card" }),
+  deleteBookmark: (id: BookmarkId) =>
+    Bookmark.actions.delete({ id, type: "card" }),
+};
+
+const connector = connect(mapStateToProps, mapDispatchToProps);
+
+type ReduxProps = ConnectedProps<typeof connector>;
+type BookmarkLoadListProps = {
+  bookmarks: IBookmark[];
+  allLoaded: boolean;
 };
 
-function QueryBuilder(props) {
+type TimelineLoadListProps = {
+  timelines: Timeline[];
+  allLoaded: boolean;
+};
+
+type QueryBuilderInnerProps = ReduxProps &
+  WithRouterProps &
+  BookmarkLoadListProps &
+  TimelineLoadListProps & {
+    route: Route;
+  };
+
+function QueryBuilderInner(props: QueryBuilderInnerProps) {
+  const isBookmarked = props.bookmarks.some(
```

Comment:
Why was this inlined? Didn't it use a selector before?
It seems like we have 2+ implementation of this logic or similar

Reply:
When I was trying to get the types to work, I was losing the types when we use the `props` in `mapStateToProps`

Reply:
I had another go at things and it looks like I was able to get the types to work once I added some types for the loaders. I think it should be good to go now